### DISCUSSION TITLE
Very slow index building

### DISCUSSION THREAD

**Discussion by chrisordio** [planetscale/discussion#646] [Q&A]
I made a Deploy Request where I added 4 indexes. In 25 hours it was only at 12%.  I thought maybe one of the indexes on a varchar column was causing the slow progress so i cancelled it and made a new Deploy Request.  This time just one index on two numeric fields in desc order.  Now it's been 4 hours at 3%.  There is zero traffic to this branch.  No data being added

The table is 68 million rows with MySQL and running a PS-900, which i think is the same as a db.m5.4xlarge in AWS RDS.  It's not a 1:1 comparison but previously using PostgreSQL with a db.m5.2xlarge on a similar table it took about an hour or two only for such an index.  Of course I was running the command direct in the psql CLI

I'm new to Planetscale and unfamiliar with what goes on under the hood? Is this speed expected? Any ideas?

**Comment by mscoutermarsh** [comment]
Hi @chrisordio,

It could be that it's being throttled to reduce impact on production traffic. Deploy requests first make a copy of the table, then apply the schema change ([full details here](https://planetscale.com/docs/learn/how-online-schema-change-tools-work)). Making the copy can take a while for large tables. Your case does sound exceptionally slow though, so there is likely something happening.

If you write into support, we'll be able to look into it in more detail and can hopefully find a solution for you!
